python/lsst/ip/diffim/detectAndMeasure.py

```diff
@@ -623,6 +630,39 @@ def calculateMetrics(self, difference):
                 self.metadata.add("%s_mask_fraction"%maskPlane.lower(), -1)
                 self.log.info("Unable to calculate metrics for mask plane %s: not in image"%maskPlane)
 
+        maglim_science = self._calculateMagLim(science)
+        fluxlim_science = (maglim_science*u.ABmag).to_value(u.nJy)
+        maglim_template = self._calculateMagLim(matchedTemplate)
+        fluxlim_template = (maglim_template*u.ABmag).to_value(u.nJy)
+        maglim_diffim = (np.sqrt(fluxlim_science**2 + fluxlim_template**2)*u.nJy).to(u.ABmag).value
+        self.metadata.add("diffimLimitingMagnitude", maglim_diffim)
+
+    def _calculateMagLim(self, exposure, nsigma=5.0):
+        """Calculate an exposure's limiting magnitude.
+
+        This method uses the photometric zeropoint together with the
+        PSF size from the center of the exposure.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            The target exposure to calculate the limiting magnitude for.
+        nsigma : `float`, optional
+            The detection threshold in sigma.
+
+        Returns
+        -------
+        maglim : `astropy.units.Quantity`
+            The limiting magnitude of the exposure.
+        """
+        zeropoint = exposure.getPhotoCalib().instFluxToMagnitude(1)
+        psf = exposure.getPsf()
```

Comment:
Thanks for flagging this. I've changed this so it will set the template limiting magnitude to `nan` if it hits this edge case. In this situation, it will adopt the (admittedly optimistic) science limiting magnitude for the diffim.